apps/domain/src/main/core/services/role_service.py

```diff
@@ -60,6 +60,15 @@ def create_role_msg(
             message="Invalid request payload, empty fields (name)!"
         )
 
+    # Check if this role name was already registered
+    try:
+        node.roles.first(name=_name)
+        raise AuthorizationError(
+            message="You can't create a new Role using this email!"
```

Comment:
Is this a "The role <name> already exists" or a "You're not authorized to update this role"?